hera_sim/components.py

```diff
@@ -1,14 +1,18 @@
 """A module providing discoverability features for hera_sim.
 """
-
-import functools
+from __future__ import annotations
 import re
-from abc import ABCMeta, abstractmethod
-
+from abc import abstractmethod
```

Comment:
Has there been a change with `abstractmethod` that I'm unaware of? At the time of writing, a class was required to have `ABCMeta` as its metaclass in order to actually use `abstractmethod` as intended. I'm not entirely convinced that the `abstractmethod` requirement is being enforced in subclasses. I tested the component discoverability with a custom class as follows:
```
from hera_sim import component

@component
class Test:
    pass

class Model(Test):
    pass
```
and this worked without a hitch, which suggests that the `abstractmethod` decorator isn't actually doing anything. (If my understanding is correct, then decorating the `__call__` method with `abstractmethod` makes it so that any subclass *must* define its own `__call__` method. If I'm wrong, then please correct me on that point.) On another note, though, the component was discovered via `get_all_components`.

Reply:
Ah. Yeah that's my mistake.